components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediatorFactory.java

```diff
@@ -0,0 +1,202 @@
+import org.apache.synapse.util.xpath.SynapseXPath;
+import org.jaxen.JaxenException;
+import org.wso2.micro.integrator.mediator.dataservice.DataServiceCallMediator.OperationsType;
+import org.wso2.micro.integrator.mediator.dataservice.DataServiceCallMediator.Operations;
+import org.wso2.micro.integrator.mediator.dataservice.DataServiceCallMediator.Param;
+
+
+import javax.xml.namespace.QName;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Factory for {@link DataServiceCallMediator} instances.
+ * <p>
+ * Configuration syntax:
+ * <pre>
+ * &lt;dataServicesCall serviceName = ""&gt;
+ *      &lt;operations type="single"/&gt;
+ *          &lt;operation name=""&gt;
+ *                  &lt;param name="string"&gt; &lt;param/&gt;
+ *          &lt;/operation&gt;
+ *      &lt;/operations&gt;
+ *      &lt;target  type="envelope"|"property" name="string"/&gt;
+ * &lt;/dataServicesCall&gt;
+ * </pre>
+ */
+
+public class DataServiceCallMediatorFactory extends AbstractMediatorFactory {
+
+    private static final QName DSSCALL_Q =
+            new QName(SynapseConstants.SYNAPSE_NAMESPACE, DataServiceCallMediatorConstants.DATA_SERVICES_CALL);
+    private static final QName TARGET_Q = new QName(
+            XMLConfigConstants.SYNAPSE_NAMESPACE, DataServiceCallMediatorConstants.TARGET);
+    private static final QName NAME_Q = new QName(DataServiceCallMediatorConstants.NAME);
+    private static final QName TYPE_Q = new QName(DataServiceCallMediatorConstants.TYPE);
+
+
+    public Mediator createSpecificMediator(OMElement elem, Properties properties) {
+
+        DataServiceCallMediator mediator = new DataServiceCallMediator();
+        processAuditStatus(mediator, elem);
+        String dsName = elem.getAttributeValue(new QName(DataServiceCallMediatorConstants.SERVICE_NAME));
+        if (dsName == null) {
+            handleException("The 'serviceName' attribute in 'dataServicesCall' element  is missing " +
+                    "in the configuration.");
+        }
+        mediator.setDsName(dsName);
+        OMElement operationsTypeElement = elem.getFirstElement();
+        if (!operationsTypeElement.getLocalName().equals(DataServiceCallMediatorConstants.OPERATIONS)) {
+            handleException("The 'operations' element in 'dataServicesCall' element  is missing in the configuration.");
+        }
+        String operationType = operationsTypeElement.getAttributeValue(new QName(DataServiceCallMediatorConstants.TYPE));
+        if (operationType == null) {
+            handleException("The 'type' attribute in 'operations' element  is missing in the configuration.");
+        }
+        OperationsType operationsType = OperationsType.valueOf(operationType.toUpperCase());
+        List operationList = extractOperations(operationsTypeElement, mediator);
+        Operations operations = mediator.new Operations(operationsType, operationList);
+        mediator.setOperations(operations);
```

Comment:
deployment should fail if we have multiple operations for  type="single_req" ?.

Reply:
No.. only the first one is considered and rest is ignored

Reply:
I still feel like failing deployment is more convenient since this is a wrong config. otherwise in runtime we will have to wonder why we have given many operations and others are not working.

Reply:
+1. I have added